app/controllers/application_controller.rb

```diff
@@ -16,6 +16,8 @@ class ApplicationController < ActionController::Base
 
   layout 'layouts/main_site'
 
+  after_action :track_ahoy_visit
```

Comment:
I'm a little apprehensive about having tracking enabled as the default for (most) controllers. It feels like an insecure default, making it more likely a sensitive controller action introduced in the distant future has its parameters logged by mistake.

Consider disabling tracking by default and only enabling on a per-controller, per-action basis, e.g.

```rb
class SomeController
  enable_tracking only: [:some_action, :another_action]
end
```

Although this is more verbose, it is the more secure default. I'd avoid having an `except:` option so developers aren't tempted to safe-list entire controllers.

Also consider disabling tracking for all logged-in admin users, regardless of whether they are using the admin backend or the public-facing site.

You may also want to sanitize the parameters that are tracked by scrubbing any params that match the names configured in `config/initializers/filter_parameter_logging.rb`.

One more thing - are any senstive tokens logged currently? E.g. password reset tokens.

Reply:
Iirc, I'm using a gem that you suggested to prevent the latter happening in production.

I wonder if I could record stats for all users, but have two views of that data - an 'end user activity only' for current_user.can?( :list, :stats ) and an 'all activity' view for current_user.can?( :list, :admin_stats ).

(The latter 'super stats admin' people should be able to view the end-user-only stats too, that's probably a useful filter to be able to choose even if you're not restricted to it.)

I was googling for options for views of this data last night and Blazer seems like the thing that gets paired with ahoy. Are there any others I should consider / any good articles I should read? I looked at chartkick but it seem similar-ish to Blazer and I couldn't find any docs for pairing it with Ahoy at first search.

Reply:
> Iirc, I'm using a gem that you suggested to prevent the latter happening in production.

I can't recall which gem this is. Something to discuss today perhaps.

> I was googling for options for views of this data last night and Blazer seems like the thing that gets paired with ahoy. Are there any others I should consider / any good articles I should read? I looked at chartkick but it seem similar-ish to Blazer and I couldn't find any docs for pairing it with Ahoy at first search.

Blazer's a good choice (one of its dependencies is chartkick).